app/models/pglogical_subscription.rb

```diff
@@ -242,8 +242,22 @@ def find_password
 
   def create_subscription
     MiqRegion.destroy_region(connection, remote_region_number)
+
+    # new_subscription_name also queries the remote, so we fetch it early to avoid a nested remote query
+    subscription = new_subscription_name
+
+    # Creating the slot and subscription enabled in the same command will hang when replicating
+    # between databases in the same cluster. Instead, we'll do the following:
+    # 1) On the remote, create a logical replication slot with pgoutput plugin name.
+    # 2) On the global, create the subscription without a slot but reference the previously created slot name.
+    # From: https://www.postgresql.org/docs/10/sql-createsubscription.html
+    with_remote_connection(5.seconds) do |conn|
+      client = PG::LogicalReplication::Client.new(conn.raw_connection)
+      client.create_logical_replication_slot(subscription)
```

Comment:
This is the part that tripped me up for a while, you can actually create the replication_slot without error on the global region (`pglogical` here) and things work a little.  But, as long as you create the objects in the right database, things work flawlessly everytime:

* replication_slot (publisher / remotes)
* publication (publisher / remotes)
* subscription (subscriber / global)

It turned out that when you had publisher and subscriber databases on different clusters/instances, the create_subscription on the subscriber side "correctly" creates the slots in each remote database.